### PR TITLE
Update petsc cmake scripts

### PR DESCRIPTION
**Pull request by endJunction**
This are the remains of the https://github.com/ufz/ogs/pull/837.

~~@bilke Attention, this cmake files overwrite the changes made by you. If OK, please merge.~~

The scripts are replaced by a submodule for sake of (hopefully) easier maintenance. The submodule originates from https://github.com/ufz/cmake-modules which is a fork from https://github.com/jedbrown/cmake-modules.

### CODE REVIEW

**Comment by endJunction** [comment]
Jenkins test this please

File: scripts/cmake/findPETSC/CorrectWindowsPaths.cmake [hunk 1]
```diff
@@ -6,8 +6,9 @@
 
 macro (CONVERT_CYGWIN_PATH _path)
   if (WIN32)
-    execute_process(COMMAND cygpath.exe -m ${${_path}}
+    EXECUTE_PROCESS(COMMAND cygpath.exe -m ${${_path}}
```
Comment (bilke) [inline scripts/cmake/findPETSC/CorrectWindowsPaths.cmake]: Please use lowercase for CMake functions.

**Comment by endJunction** [comment]
Not sure why travis can not checkout the new submodule...

**Comment by bilke** [comment]
I guess when cloning with the git protocol you need to have a SSH key set up. Maybe you can change to `https` instead.

**Comment by bilke** [comment]
Does not work yet on Mac. I will solve this later on.

**Pull request merged**